server/graphql/v1/mutations/orders.js

```diff
@@ -184,9 +184,10 @@ const checkAndUpdateProfileInfo = async (order, fromAccount, isGuest, currency)
   const { totalAmount, fromAccountInfo, guestInfo } = order;
   const accountUpdatePayload = {};
   const location = fromAccountInfo?.location || guestInfo?.location || fromAccount.location;
+  const isContributingFromSameHost = fromAccount.HostCollectiveId === order.collective.HostCollectiveId;
 
   // Only enforce profile checks for guests and USD contributions at the moment
-  if (isGuest && currency === 'USD') {
+  if (isGuest && currency === 'USD' && !isContributingFromSameHost) {
```

Comment:
This is cosmetic as `isGuest` will be false when contributing with an existing profile, but I wanted to make sure we keep track of this condition in the API